File: olimpiadas II - 20232/Maraton Tech/Java/src/parindromo.java
import java.util.Scanner;

public class parindromo {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		while (sc.hasNextLine()) {
			String cad = sc.nextLine();
			cad = cad.replace(" ", "");
			System.out.println(getParindromoResult(cad));
		}
		sc.close();
	}

	private static String getParindromoResult(String cad) {
		final int len = cad.length();
		String result = "None";
		for (int i = 0; i < len - 2; i++) {
			for (int j = i + 2; j <= len; j++) {
				final String sub = cad.substring(i, j);
				if (isPalindrome(sub)) {
					if (result.equals("Odd") && sub.length() % 2 == 0)
						return "Both";
					else if (result.equals("Even") && sub.length() % 2 == 1)
						return "Both";
					else if (result.equals("None"))
						result = sub.length() % 2 == 0 ? "Even" : "Odd";
				}
			}
		}
		return result;
	}

	private static boolean isPalindrome(String cad) {
		return new StringBuilder(cad).reverse().toString().equals(cad);
	}
}

File: olimpiadas II - 20232/Maraton Tech/Python/parindromo.py
def isPalindrome(cad):
    return cad == cad[::-1]

def getPalindromeResult(cad):
    result = "None"
    for i in range(len(cad) - 2):
        for j in range(i + 2, len(cad) + 1):
            sub = cad[i:j]
            if isPalindrome(sub):
                if result == "Odd" and len(sub) % 2 == 0:
                    return "Both"
                elif result == "Even" and len(sub) % 2 == 1:
                    return "Both"
                elif result == "None":
                    if len(sub) % 2 == 0:
                        result = "Even" 
                    else:
                        result = "Odd"
    return result

cad = input()
cad = cad.replace(" ", "")
print(getPalindromeResult(cad))